docs/DeveloperGuide.md

```diff
@@ -1,34 +1,234 @@
+    + [2.2.2. Favourite command description modifier (`/descfav` Feature)](#222-favourite-command-description-modifier-descfav-feature)
+- [3. Appendix I: Requirements](#3-appendix-i-requirements)
+  * [3.1 Product scope](#31-product-scope)
+    + [3.1.1 Target user profile](#311-target-user-profile)
+    + [3.1.2 Value](#312-value)
+  * [3.2. User Stories](#32-user-stories)
+  * [3.3. Non-Functional Requirements](#33-non-functional-requirements)
+  * [3.4. Glossary](#34-glossary)
+- [4. Appendix II: Instructions for manual testing](#4-appendix-ii-instructions-for-manual-testing)
 
-## Product scope
-### Target user profile
+## 1. Setting up, getting started
 
-{Describe the target user profile}
+Refer to the guide [Setting up and getting started](https://github.com/AY2021S1-CS2113T-F14-3/tp/blob/master/README.md#duke-project-template).
 
-### Value proposition
+## 2. Design & Implementation
 
-{Describe the value proposition: what problem does it solve?}
+This section describes the design and implementation of the product. It has been divided into two sections: Architecture 
+and Implementation. 
 
-## User Stories
+### 2.1. Architecture
 
-|Version| As a ... | I want to ... | So that I can ...|
+The architecture diagram given in the figure below explains the high-level design of the App. 
+
+![Architecture Diagram](DG_Diagrams/Architecture.png)
+
+Given below is a quick overview of each component.
+
+`Main` is the class Duke.java. It is responsible for:
+
+- Initializing the components in the correct sequence, and calling or creating objects in the order of execution.
+
+The rest of the App consists of 4 main components:
+
+- `Ui` : The Ui component handles all interactions with the user.
+- `Logic` : The Logic component makes sense of the command and executes it.
+- `Model` : The Model component is responsible for all data held in the memory.
+- `Storage` : The Storage component handles data by reading from and writing to files in the hard disk. 
+
+The sections below give more details of each component.
+
+#### 2.1.1 Ui Component
+The Ui component is responsible for getting the user input and for displaying appropriate messages as response 
+to commands executed by the logic component. The following class diagram gives a brief overview of the Ui component 
+and it's interaction with other components.
+
+![uicomponent](DG_Diagrams/UiComponent.png)
+
+#### 2.1.2. Logic Component
+The Logic component is responsible for the following tasks:
+- Makes sense of the command entered by the user by splitting it into the command type and the parameters and/or 
+delimiters if any.
+- Executes a command based on its type and provides an appropriate result to the user via the Ui component.
+
+The following class diagram briefly explains how different classes in the Logic component interact with each other.
+
+<img src="DG_Diagrams/LogicComponent.png" alt="logiccomponent" width=1100 height=500>
```

Comment:
Perhaps splitting this diagram into smaller portions like the Model component will be good